### PR TITLE
Update cluster-api to v1.4.0-beta.2

### PR DESCRIPTION
**Pull request by Amulyam24**
**What this PR does / why we need it**:
Update cluster-api to v1.4.0-beta.2

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

/area provider/ibmcloud

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note
Update cluster-api to v1.4.0-beta.2
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -18,36 +18,35 @@ require (
...
 	github.com/stretchr/testify v1.8.1
 	golang.org/x/crypto v0.7.0
 	golang.org/x/text v0.8.0
-	k8s.io/api v0.25.0
-	k8s.io/apiextensions-apiserver v0.25.0
-	k8s.io/apimachinery v0.25.0
+	k8s.io/api v0.26.1
+	k8s.io/apiextensions-apiserver v0.26.1
+	k8s.io/apimachinery v0.26.1
 	k8s.io/cli-runtime v0.25.0
```
Comment (mkumatag) [inline go.mod:32]: do we need to bump this as well?

**Review by mkumatag — CHANGES_REQUESTED** [review]
one minor comment, otherwise lgtm

File: go.mod (hunk 1 above)
Comment (Amulyam24) [reply go.mod:32]: @mkumatag, cluster-api 1.4.x is using v0.26.1 and this change has upgraded these here.

Comment (mkumatag) [reply go.mod:32]: I meant for `k8s.io/cli-runtime` which is still pointing to `v0.25.0`

Comment (Amulyam24) [reply go.mod]: > I meant for `k8s.io/cli-runtime` which is still pointing to `v0.25.0`

Oops, missed it. Updated, PTAL!

**Pull request merged**